### PR TITLE
Add scaling of DIB and DISMF by running mean of removed Antarctic ice runoff 

### PR DESCRIPTION
**Pull request by xylar**
This merge adds a capability by which data iceberg (DIB) freshwater fluxes and data ice-shelf melt fluxes (DISMF) can be scaled by the running mean of the the amount of ice (solid) runoff coming from Antarctica (which is removed from the system by MPAS-Ocean).  

This merge includes:
* A capability in MPAS-Ocean to compute daily averages of the area integrated removed ice runoff
* A capability in MPAS-Ocean to compute the running mean of these daily means over a given period of time (currently 2 years = 730 days)
* A capability to pass this running mean through the coupler (via the field `rmean_rmv_ice_runoff` in the `infodata` datastructure) from MPAS-Ocean to MPAS-Seaice
* A requirement that the  the total annual mean DIB and DISMF fluxes be computed and stored in the DIB and DISMF input files (fulfilled by https://github.com/MPAS-Dev/compass/pull/836).  These files are currently only available for the IcoswISC30E3r5 mesh.
* Capabilities to compute a scale factor between the annual mean total data freshwater flux and the running mean of the removed ice runoff in both MPAS-Ocean and MPAS-Seaice
* A capability to scale `landIceFreshwaterFlux` and `landIceHeatFlux` in MPAS-Ocean by the scale factor
* A capability to scale `bergFreshwaterFlux` and `bergLatentHeatFlux` in MPAS-Seaice by the scale factor

[NML]
[BFB] stealth feature

------

The capability is described in more detail in this design document
https://acme-climate.atlassian.net/wiki/spaces/PSC/pages/4210098268/Design+Document+Data+iceberg+and+ice-shelf+melt+flux+patterns+for+E3SM+runs

Relevant discussion can be seen at https://github.com/E3SM-Ocean-Discussion/E3SM/pull/109

### CODE REVIEW

File: cime_config/tests.py [hunk 1]
```diff
@@ -269,6 +269,7 @@
             "SMS_D_Ld1.T62_oQU240.GMPAS-IAF.mpaso-harmonic_mean_drag",
             "SMS_D_Ld1.T62_oQU240.GMPAS-IAF.mpaso-upwind_advection",
             "ERS_Ld5_D.T62_oQU240.GMPAS-IAF.mpaso-conservation_check",
+            "ERS.ne30pg2_r05_IcoswISC30E3r5.CRYO1850-DISMF.mpaso-scaled_dib_dismf",
```
Comment (xylar) [inline cime_config/tests.py:272]: Probably need to reduce the duration:
```suggestion
            "ERS_Ld5.ne30pg2_r05_IcoswISC30E3r5.CRYO1850-DISMF.mpaso-scaled_dib_dismf",
```

**Comment by xylar** [comment]
This is based off of #6693, and needs those changes to work correctly in G-case configurations.

File: components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/scaled_dib_dismf/README [hunk 2]
```diff
@@ -0,0 +1,8 @@
+This testdef is used to test a stealth feature in mpaso and mpassi introduced
+by PR #XXXX. It simply changes sets the following namelist option in mpaso
```
Comment (xylar) [inline components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/scaled_dib_dismf/README:2]:
```suggestion
by PR #6696. It simply changes sets the following namelist option in mpaso
```

**Comment by cbegeman** [comment]
@xylar Is there any testing that would be helpful for me to do?

**Comment by xylar** [comment]
@cbegeman, I think we're mostly waiting on the longer run that @darincomeau is doing.  I can't think of other testing that I didn't already report on https://github.com/E3SM-Ocean-Discussion/E3SM/pull/109, at least not anything that would be a good use of your time.

**Comment by darincomeau** [comment]
Update, longer test run is in year 50 now: https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.dcomeau/E3SMv3_dev/20241018.v3.SORRME3r3.piControl.scaledDISMF.chrysalis/mpas_analysis/ts_0001-0050_climo_0041-0050/

Note ISMF is not plotting due to https://github.com/E3SM-Project/E3SM/issues/6719, fixed in https://github.com/E3SM-Project/E3SM/pull/6729 
Plan is to run it 100 years to see longer trends in sea level/AIS mass imbalance, but looking promising so far.

**Comment by xylar** [comment]
@jonbob, if you look at the surface fluxes, a lot of them show atmosphere blockiness.  Is that expected?  Here's an example:
image
Less so here:
image

Not related to this PR but showing up in the analysis above.

**Comment by xylar** [comment]
In terms of mass imbalance and ssh, no sign of trouble:
image
image

But we need more years to be sure, so, yes, lets do the full 100 years.

**Comment by jonbob** [comment]
@xylar -- @cbegeman and I were looking at this last week. We rely on the nonlinear CAAS mapping files to smooth out some of the grid imprinting, but we found that snow and rain are removed from the list of fluxes that get this treatment. It's easily changed in user_nl_cpl if you want to test the impact.

**Comment by jonbob** [comment]
And I guess the first thing to do is to make sure we have the nonlinear mapping file for atm2ocn specified for this run

**Comment by cbegeman** [comment]
@xylar and @jonbob I discussed this with @proteanplanet and we will not be changing the atmosphere mapping files because it would compromise conservation accuracy across the coupled system, which is a higher priority. Thus, we will be retaining the blockiness moving forward.

**Comment by proteanplanet** [comment]
Added information: The decision was made to switch off non-linear remapping after following testing in the lead-up to v3, but just for mass fluxes as documented by Andrew Bradley in the Critical Path meeting on December 4, 2023:
[https://acme-climate.atlassian.net/wiki/spaces/CM/pages/4021780481/2023-12-04+v3+Critical+Path+Meeting](https://acmeclimate.slack.com/archives/G01HC3ZU746/p1730486812327149)
The subsequent pull request reverting non-linear mass flux remapping is here: https://github.com/E3SM-Project/E3SM/pull/6094

**Comment by rljacob** [comment]
which grid set is this?  I think we only automatically turn on CAAS for the v3 grid set.

**Comment by jonbob** [comment]
@rljacob -- it's using the new SOwISC12to30E3r3 grid

**Comment by xylar** [comment]
@jonbob (and anyone else who wants to chime in), is there a reason we can't use higher order, conservative (linear) mapping files from TempestRemap for these fields?  It would seem to me that would reduce blockiness without compromising conservation.  I'm mostly just seeking to understand why some flux fields are blocky and others don't seem to be.

**Comment by jonbob** [comment]
@xylar -- we do use conservative mapping files from tempest for all flux fields. The nonlinear CAAS mapping files are also used for some, but not all, of those fluxes. We would probably see less blockiness by also using the nonlinear mapping files for snow and rain -- but apparently there has been a decision not to do that

**Comment by rljacob** [comment]
Some discussion here: https://acme-climate.atlassian.net/wiki/spaces/CM/pages/4016177153/2023-12-01+v3+Critical+Path+Meeting#Discussion-on-coupler-water-budgets
The nonlinear maps plus the way the coupler does fractions leads to water budget issues for tri-grid.

**Comment by proteanplanet** [comment]
The links @rljacob and I have provided above summarize the issue, but ultimately it boils down to several decimal places of accuracy in the global water budget; non-linear remapping leaks mass more than linear remapping.  

To answer your questions @xylar: A reason why certain fields in sea ice show the aliasing, but others do not, is that sea ice growth is particularly sensitive to snow conductivity.  There are other reasons associated with the aging of snow, too, but rain does not play a role because it just runs off.

**Comment by xylar** [comment]
Okay, sorry for the noise about this.  It doesn't relate to this PR in any case, I just noticed it and it looked pretty ugly.

**Comment by proteanplanet** [comment]
> Okay, sorry for the noise about this. It doesn't relate to this PR in any case, I just noticed it and it looked pretty ugly.
I'm very glad you noticed it and questioned it.  The more scrutiny E3SM receives, the better it gets.

**Comment by xylar** [comment]
@cbegeman and I feel uncomfortable with the amount of long-period (~20 year) oscillation in the cumulative land-ice mass removed:
image

@cbegeman suggested reducing the duration of the running mean to 2 years instead of 20, so we are going to start a new simulation with that change.

I have also rebased to pick up recent changes (including the fix to total land ice fluxes #6729) and cherry-picked the commits from #6501 so we should no longer have redundant frazil fluxes under ice shelves.

**Comment by xylar** [comment]
@mahf708, do you have any intuition about why my rebase caused a bunch of CI failures in tests that seem to have nothing to do with the changes here?

**Comment by xylar** [comment]
@mahf708, ignore that.  I committed some submodule changes by mistake and that's almost certainly what's wrong.

**Comment by darincomeau** [comment]
>@cbegeman suggested reducing the duration of the running mean to 2 years instead of 20, so we are going to start a new simulation with that change.

This run with the 2 year running mean is in year 15 (plan to run to 30), so far looking as expected.

**Comment by darincomeau** [comment]
30 year run with 2 year running mean complete, analysis for last decade: https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.dcomeau/E3SMv3_dev/20241115.v3.SORRME3r3.piControl.scaledDISMF.chrysalis/mpas_analysis/ts_0001-0030_climo_0021-0030/

image
image
image

This all looks good to me, glad @cbegeman and @xylar suggested the shorter running mean.

**Review by darincomeau — APPROVED** [review]
Approved based on test simulations. Thanks @xylar for all the work here, this is really nice to have!

**Comment by darincomeau** [comment]
Also, here are coupler budgets (for all time) at end of 30 year run here:

```
(seq_diag_print_mct) NET WATER BUDGET (kg/m2s*1e6): period = all_time: date =    310101     0
                           atm            lnd            rof            ocn         ice nh         ice sh            glc        *SUM*
         wfreeze     0.00000000     0.00000000     0.00000000    -0.15841287     0.10203609     0.05637678     0.00000000    -0.00000000
           wmelt     0.00000000     0.00000000     0.00000000     0.62231655    -0.27449790    -0.34781891     0.00000000    -0.00000027
           wrain   -32.21941172     7.12336826     0.00000000    25.03161382     0.04754609     0.01606129     0.00000000    -0.00082227
           wsnow    -2.35181154     0.89989044     0.00000000     0.99777569     0.14355284     0.31029713     0.00000000    -0.00029545
          wpolar     0.00000000     0.00000000     0.00000000     0.00024278     0.00000000     0.00000000     0.00000000     0.00024278
           wevap    34.56863999    -5.27370184     0.00000000   -29.25959786    -0.01383098    -0.02163831     0.00000000    -0.00012900
         wrunoff     0.00000000    -2.59830220     0.00353440     2.59474683     0.00000000     0.00000000     0.00000000    -0.00002097
         wfrzrof     0.00000000    -0.15479011     0.00000000     0.15479136     0.00000000     0.00000000     0.00000000     0.00000125
          wirrig     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000
           *SUM*    -0.00258327    -0.00353545     0.00353440    -0.01652369     0.00480613     0.01327798     0.00000000    -0.00102391

 ```

and from end of 30 year run with `CRYO1850` compset:

```
(seq_diag_print_mct) NET WATER BUDGET (kg/m2s*1e6): period = all_time: date =    310101     0
                           atm            lnd            rof            ocn         ice nh         ice sh            glc        *SUM*
         wfreeze     0.00000000     0.00000000     0.00000000    -0.14076901     0.09760558     0.04316343     0.00000000     0.00000000
           wmelt     0.00000000     0.00000000     0.00000000     0.55585636    -0.27026780    -0.28558717     0.00000000     0.00000139
           wrain   -32.89018544     7.20408830     0.00000000    25.62460457     0.04388258     0.01681237     0.00000000    -0.00079762
           wsnow    -2.14681999     0.90224016     0.00000000     0.85610971     0.13678902     0.25133812     0.00000000    -0.00034300
          wpolar     0.00000000     0.00000000     0.00000000     0.02162830     0.00000000     0.00000000     0.00000000     0.02162830
           wevap    35.03631225    -5.32849725     0.00000000   -29.67873015    -0.01295414    -0.01624372     0.00000000    -0.00011300
         wrunoff     0.00000000    -2.61771739     0.00374754     2.61394829     0.00000000     0.00000000     0.00000000    -0.00002156
         wfrzrof     0.00000000    -0.16767395     0.00000000     0.16767497     0.00000000     0.00000000     0.00000000     0.00000101
          wirrig     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000
           *SUM*    -0.00069318    -0.00756014     0.00374754     0.02032304    -0.00494476     0.00948303     0.00000000     0.02035554
          
 ```
           
so the `wpolar` term (due to AIS mass imbalance) is significantly reduced here down to magnitude of other terms.

**Review by cbegeman — APPROVED** [review]
Approving based on code inspection and decadal testing showing low mass imbalance. I'm really happy the shorter averaging time worked out so well. Thanks so much @xylar for your hard work getting this feature in!

**Comment by rljacob** [comment]
Its not clear from the PR description why this is non-BFB.

**Comment by darincomeau** [comment]
Good point @rljacob - I actually don't think it is non-BFB. It's certainly BFB for all non-`*-DISMF` cases, and the default namelist options set here (e.g. `config_scale_dismf_by_removed_ice_runoff=.false.`) should keep it BFB even in those cases.

@xylar do you agree this is [BFB], or am I missing something?

**Comment by jonbob** [comment]
Passes (with expected NML DIFFs):
* ERP_Ld3.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel.allactive-pioroot1
* SMS_D_Ld1.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel.allactive-wcprod

Also successfully ran the new test on chrysalis:
* ERS_Ld5_PS.ne30pg2_r05_IcoswISC30E3r5.CRYO1850-DISMF.chrysalis_intel.mpaso-scaled_dib_dismf

**Comment by jonbob** [comment]
merged to next

**Comment by jonbob** [comment]
merged to master and expected NML DIFFs blessed, except for pm-cpu which has yet to report

note that we had a few failures on non-lcrc platforms due to the new iceberg climatology file not being staged on the inputdata server. Those should clear with the next round of testing, though more blesses may be required after that

**Pull request merged**